controllers/core/configmap_controller.go

```diff
@@ -67,11 +68,27 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// Check if the flag value has changed
 	newWinIPAMEnabledCond := r.Condition.IsWindowsIPAMEnabled()
```

Comment:
Loos like the ConfigMap fetched at line 56 is not used ... Can you make the change to pass the configmap to `IsWindowsIPAMEnabled` and the new function `IsWindowsPrefixDelegationEnabled` instead of calling the API again over there? Thanks.

Reply:
If the configmap is not existing, the controller will either return earlier or return an error. Let's keep them as they are for now. Calling cached client is cheap. Making another call make help avoiding outdated data. We can revisit this later anyway.